test/fakes/fakedatastore/fakedatastore.go

```diff
@@ -619,6 +629,10 @@ func (s *DataStore) PruneJoinTokens(ctx context.Context, req *datastore.PruneJoi
 	return &datastore.PruneJoinTokensResponse{}, nil
 }
 
+func (s *DataStore) WithError(err error) {
```

Comment:
nit: `With*` functions generally return something that has been embellished "with" something
```suggestion
func (s *DataStore) SetError(err error) {
```